org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundIterator.java

```diff
@@ -56,6 +56,10 @@ private IPosition createFirstPosition() throws GeneratorException {
 	private IPosition next;
     private boolean justDidNext = false; // This attempts to deal with the case where someone does .next() without .hasNext();
 
+    // TODO: fix hasNext to only test if there is another element, hasNext should not increment the index
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)